asab/application.py

```diff
@@ -602,18 +606,31 @@ def _set_housekeeping_time_from_config(self):
 		# if the app started after the housekeeping time, set it to the next day
 		if now > next_housekeeping_time:
 			next_housekeeping_time += datetime.timedelta(days=1)
-		return next_housekeeping_time
+
+		# compute the time limit for the housekeeping
+		time_delta_limit = config_time_limit - config_house_time
+		if time_delta_limit < datetime.timedelta(hours=0):
+			time_delta_limit += datetime.timedelta(days=1)
+
+		next_time_limit = next_housekeeping_time + time_delta_limit
+
+		return (next_housekeeping_time, next_time_limit)
 
 	def _on_housekeeping_tick(self, message_type):
 		"""Check if it's time for 'Application.housekeeping!'.
-		If so, publish the message and set housekeeping time for the next day.
+		If so, publish the message and set housekeeping time and the time limit for the next day.
 		"""
-		if datetime.datetime.now(datetime.timezone.utc) > self.NextHousekeeping:
-			self.PubSub.publish("Application.housekeeping!")
-			self.NextHousekeeping += datetime.timedelta(days=1)
+		now = datetime.datetime.now(datetime.timezone.utc)
+		if self.HousekeepingTime < now:
+			if now < self.HousekeepingTimeLimit:
+				self.PubSub.publish("Application.housekeeping!")
+			else:
+				L.warning("Housekeeping has not been performed. It is past the time limit.")
+			self.HousekeepingTime += datetime.timedelta(days=1)
```

Comment:
but what if your computer has been asleep for two or more days?

Reply:
> one nitpicky question - what if you start the app between the housekeeping start time and the limit time? should the housekeeping be triggered right away? or the next day?

The next day. As it was before. The limit time only prevents accidental housekeeping like in the scenario mentioned above.

Reply:
> but what if your computer has been asleep for two or more days?

Maybe instead of adding 24 hours we could check again call `_set_housekeeping_time_from_config()` - that is, do the same machinery as in the app init - set housekeeping and limit time for tomorrow from now.

Does configparser in Python load the configuration file only once at the beginning of the application, and will subsequent calls to access the configuration data always retrieve the same data, regardless of any changes made to the configuration file in the meantime? I think the answer is _yes_. I'm I right, @byewokko ?

Reply:
ASAB don't do *intentionally* dynamic config reload. The strategy is to restart the microservice.

Reply:
and for a housekeeping:

Each housekeeping has its unique number - a number of a day eg. since the service start
Housekeeping component remembers the last number.

If it receive on_tick event within the defined time window AND the "last number" is lower than the current, it executes housekeeping and updates "last number".

This algo will prevent:
1) housekeeping outside of the time window
2) double execution of the housekeeping